odoo/fields.py

```diff
@@ -1396,9 +1396,6 @@ def convert_to_column(self, value, record, values=None, validate=True):
         return int(value or 0)
 
     def convert_to_cache(self, value, record, validate=True):
-        if isinstance(value, dict):
-            # special case, when an integer field is used as inverse for a one2many
-            return value.get('id', None)
```

Comment:
Please check again.  I believe it may happen in this situation:
* a form view contains a one2many field;
* the inverse of the one2many field is an integer field (instead of a many2one field);
* an onchange is triggered on a line inside the one2many field.

When this happens, I think that the client sends the main record's data as a `dict` under the one2many's inverse field.  I don't know whether we actually use such a case, but if it is possible, some developer will abuse it someday :smile:

Reply:
You are perfectly right :+1: . I tested it again this morning and it crashes.
-> I removed this last commit

Reply:
> You are perfectly right +1 . I tested it again this morning and it crashes. -> I removed this last commit

could you least add a test that describes this feature so we don't make the same mistake ?